website/docs/r/nsxt_edgegateway.html.markdown

```diff
@@ -223,12 +272,29 @@ definition at provider level.
 * `description` - (Optional) A unique name for the edge gateway.
 * `external_network_id` - (Required) An external network ID. **Note:** Data source [vcd_external_network_v2](/providers/vmware/vcd/latest/docs/data-sources/external_network_v2)
 can be used to lookup ID by name.
-* `subnet` - (Required) One or more [subnets](#edgegateway-subnet) defined for edge gateway.
 * `edge_cluster_id` - (Optional) Specific Edge Cluster ID if required
 * `dedicate_external_network` - (Optional) Dedicating the External Network will enable Route Advertisement for this Edge Gateway. Default `false`.
 
+* `subnet` - (Optional) One or more [subnets](#edgegateway-subnet) defined for edge gateway. One of
+  `subnet`, `auto_subnet` or `auto_allocated_subnet` is **required**.
+* `auto_subnet` - (Optional, *v3.9+*) One or more [subnets](#edgegateway-auto-subnet) defined for
+  Edge Gateway. One of `subnet`, `auto_subnet` or `auto_allocated_subnet` is **required**.
+* `auto_allocated_subnet` - (*v3.9+*) One or more [subnets](#edgegateway-auto-allocated-subnet)
+  defined for Edge Gateway. One of `subnet`, `auto_subnet` or `auto_allocated_subnet` is
+  **required**.
+* `total_allocated_ip_count` - (Optional, *v3.9+*) Required with `auto_subnet`
```

Comment:
Looks like we need to state that `total_allocated_ip_count` is to be used only with `auto_subnet`. Right now it reads as if you can use it in all three subnet types, but it's required for `auto_subnet`.

Reply:
I have improved that comment.